dpctl/_sycl_platform.pyx

```diff
@@ -175,10 +175,43 @@ cdef class SyclPlatform(_SyclPlatform):
                 "a SYCL filter selector string."
             )
 
-    def print_platform_info(self):
+    def print_platform_info(self, verbosity=0):
         """ Print information about the SYCL platform.
+
+        The level of information printed out by the function can be controlled
+        by the optional ``vebosity`` setting.
+
+            - **Verbosity level 0**: Prints out the list of platforms and their
+              names.
+            - **Verbosity level 1**: Prints out the name, version, vendor,
+              backend, number of devices for each platform.
+            - **Verbosity level 2**: At the highest level of verbosity
+              everything in the previous levels along with the name, version,
+              and filter string for each device is printed.
+
+        Args:
+            verbosity (optional): Defaults to 0.
+                The verbosity controls how much information is printed by the
+                function. 0 is the lowest level set by default and 2 is the
+                highest level to print the most verbose output.
+
         """
-        DPCTLPlatformMgr_PrintInfo(self._platform_ref)
+        cdef size_t v = 0
+
+        if not isinstance(verbosity, int):
+            print(
```

Comment:
Using ``warnings.warn("warning_text")`` is more appropriate here.

Reply:
Done.